otter/convergence/transforming.py

```diff
@@ -0,0 +1,117 @@
+"""
+Functions for transforming collections of steps.
+
+There are a few classes of transformations here:
+
+- optimizing (converting multiple steps into one for the purposes of reducing
+  API roundtrips)
+- limiting (by truncating the number of steps we take)
+- throttling (by wrapping multiple steps with one that gradually executes them)
+"""
```

Comment:
This is probably just bike-shedding :) but it feels like you are arranging the code based on its _type_ rather than its _role_. Here, this module does more than one thing (role) but is in same module because of its type (functional non-effect code). IMHO, the code should be organized based on its role. So, here the optimizing part can remain in `plan.py` and rest (limiting and throttling) in say `throttle.py`. I do like the idea of representing throttle as a separate step :). In any case, this is non-blocking.